lib/mongoid/criterion/scrollable.rb

```diff
@@ -19,7 +19,9 @@ def scroll(cursor = nil, &_block)
         cursor = cursor.is_a?(Mongoid::Scroll::Cursor) ? cursor : Mongoid::Scroll::Cursor.new(cursor, cursor_options)
         # scroll
         if block_given?
-          criteria.where(cursor.criteria).order_by(_id: scroll_direction).each do |record|
+          combo_criteria = criteria.klass.and(criteria.selector, cursor.criteria)
+          combo_criteria.options = criteria.options
+          combo_criteria.order_by(_id: scroll_direction).each do |record|
```

Comment:
I almost want to call this `cursor_criteria` to be consistent, up to you if you feel it's better/worse.